server-kotlin/src/main/kotlin/digital/cesko/internet_stream/InternetStreamService.kt

```diff
@@ -0,0 +1,200 @@
+ * budgets.baseUrls - url of open data folder for one city
+ * budgets.fileUrls - name of zipped file in .csv file which contains budget
+ * budgets.insertProfileIds - id of profileId into which insert parsed data from .csv, first entry is for city etc..
+ * example of complete url: http://rozpocet.mestocernosice.cz/opendata/opendata_2020_CSV.zip
+ */
+
+@Service
+class InternetStreamService() {
+    val csvFormat = CSVFormat.DEFAULT.withDelimiter(';').withFirstRecordAsHeader()
+            .withIgnoreHeaderCase().withTrim()
+
+    @Value("\${budgets.baseSourceUrls}") val baseUrls: MutableList<String> = mutableListOf()
+    @Value("\${budgets.fileUrls}") val fileUrls: MutableList<String> = mutableListOf()
+    @Value("\${budgets.cityUrls}") val cityUrls: MutableList<String> = mutableListOf()
+    val bufferSize = 8192
+
+    @Scheduled(fixedRateString = "\${budgets.frequency}")
+    fun fetchData() {
+        val profileIds = getProfilesIds()
+
+        for ((index,baseUrl) in baseUrls.withIndex()) {
+            val completeBudgetPerCity: MutableList<Budget> = mutableListOf()
+            val cityId = profileIds[index]
+            for (fileUrl in fileUrls) {
+                var zipInputStream: ZipInputStream? = null
+                var downloadInputStream: InputStream? = null
+                val budgetPerYear: MutableList<Budget> = mutableListOf()
+                var url: String? = null
+                try {
+                    url = baseUrl + fileUrl
+                    val downloadUrl = URL(url)
+                    downloadInputStream = downloadUrl.openStream()
+                    zipInputStream = ZipInputStream(downloadInputStream)
+                    var ze: ZipEntry? = zipInputStream.nextEntry
+                    while (ze != null) {
+                        if (ze.name == "RU.csv" || ze.name == "SK.csv") {
+                            val byteArray = unzipFromStream(zipInputStream)
+                            val budget = parseBudgets(byteArray)
+                            budgetPerYear.addAll(budget)
+                        }
+                        ze = zipInputStream.nextEntry
+                    }
+                    completeBudgetPerCity.addAll(budgetPerYear)
+                } catch (malformedUrl: MalformedURLException) {
+                    logger.error("Unable to download budgets from $url", malformedUrl)
+                } catch (ioException: IOException) {
+                    logger.error("Unable to process budgets from $url", ioException)
+                } finally {
+                    zipInputStream?.close()
+                    downloadInputStream?.close()
+                }
+            }
+            deleteCityBudgets(cityId)
+            saveCityBudgets(cityId, completeBudgetPerCity)
+            completeBudgetPerCity.clear()
+        }
+    }
+
+    // Get profile_id of every city defined in application.properties.
+    // Search is made by column url.
+    fun getProfilesIds(): MutableList<Int> {
```

Comment:
Tahle funkce ma nekolik zajimavych featur.
1. Bere si data z globalni promenne cityUrls => bylo by citelnejsi, kdyby dostala cityUrls jako parametr.
2. Volajici kod pocita s tim, ze vysledny list vrati profily 1:1 (pocet, poradi), coz neni uplne intuitivni. Pro volajiciho by se to pouzivalo lepe, kdyby funkce vracela `List<Pair<String, Int>>` tedy `(cityUrl, profileId)`.